File: .trash/mapreduce_framework/wordcount_reducer.cpp
#include <iostream>
#include <string>
#include <sstream>

int main() {
    std::string line;
    std::string current_word;
    int current_count = 0;
    
    while (std::getline(std::cin, line)) {
        if (line.empty()) continue;
        
        std::istringstream iss(line);
        std::string word;
        int count;
        
        if (iss >> word >> count) {
            if (current_word == word) {
                current_count += count;
            } else {
                if (!current_word.empty()) {
                    std::cout << current_word << "\t" << current_count << std::endl;
                }
                current_word = word;
                current_count = count;
            }
        }
    }
    
    // Output the last word
    if (!current_word.empty()) {
        std::cout << current_word << "\t" << current_count << std::endl;
    }
    
    return 0;
}

File: .trash/mapreduce_framework/wordcount_reducer.py
#!/usr/bin/env python3
"""
Word Count Reducer - Reads sorted key-value pairs from stdin, sums counts
Input format: word\t1
Output format: word\tcount
"""
import sys

def main():
    current_word = None
    current_count = 0
    
    for line in sys.stdin:
        line = line.strip()
        if not line:
            continue
            
        # Parse word and count
        parts = line.split('\t')
        if len(parts) != 2:
            continue
            
        word, count = parts[0], int(parts[1])
        
        if current_word == word:
            current_count += count
        else:
            if current_word:
                print(f"{current_word}\t{current_count}")
            current_word = word
            current_count = count
    
    # Output the last word
    if current_word:
        print(f"{current_word}\t{current_count}")

if __name__ == "__main__":
    main()
